app/Http/Controllers/PostController.php

```
@@ -79,15 +80,93 @@ private function broadcastNotification($notification, $sender, $post, $type): vo
+        $user = Auth::user();
+        if (!$user) {
+            return $this->respondWithMessage($request, 'Unauthorized', false, 401);
+        }
+
+        $request->validate([
+            'description' => 'nullable|string|max:5000',
+        ]);
+
+        $original = Post::with('user')->findOrFail((int) $postId);
+
+        $repost = Post::create([
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$repost'. (undefined)

(UnusedLocalVariable)